webpack_loader/templatetags/webpack_loader.py

```diff
@@ -7,15 +7,22 @@
 register = template.Library()
 
 
-@register.simple_tag
-def render_bundle(bundle_name, extension=None, config='DEFAULT', suffix='', attrs='', is_preload=False):
+@register.simple_tag(takes_context=True)
+def render_bundle(context, bundle_name, extension=None, config='DEFAULT', suffix='', attrs='', is_preload=False, skip_common_chunks=False):
     tags = utils.get_as_tags(
         bundle_name, extension=extension, config=config,
         suffix=suffix, attrs=attrs, is_preload=is_preload
     )
+    used_tags = context.get("webpack_loader_used_tags", [])
+    if not used_tags:
+        context["webpack_loader_used_tags"] = []
+    if skip_common_chunks:
+        tags = [mark_safe(tag) for tag in tags if tag not in used_tags]
+
+    context["webpack_loader_used_tags"] = context["webpack_loader_used_tags"] + tags
```

Comment:
I think we can use sets here, like:
```python
    if "webpack_loader_used_tags" not in context:
        context["webpack_loader_used_tags"] = set()
    used_tags = context["webpack_loader_used_tags"]
    if skip_common_chunks:
        tags = [mark_safe(tag) for tag in tags if tag not in used_tags]
    context["webpack_loader_used_tags"].update(tags)
```

Reply:
I commented the same too, but removed it shortly after: unfortunately, the assumption is wrong. As far as I know, `set()` only preserves order as of python3.6.

Rendering tags in the wrong order can be a problem.

Reply:
But isn't the set only used for checking the duplicates?

Reply:
Oh, that's true. I take it back then.

Reply:
This is changed to use sets now.